interface/ceed-operator.c

```diff
@@ -640,8 +640,9 @@ int CeedOperatorSetField(CeedOperator op, const char *field_name, CeedElemRestri
             "ElemRestriction with %" CeedInt_FMT " elements incompatible with prior %" CeedInt_FMT " elements", num_elem, op->num_elem);
 
   CeedInt num_qpts = 0;
-  CeedCall(CeedBasisGetNumQuadraturePoints(b, &num_qpts));
-  CeedCheck(b == CEED_BASIS_COLLOCATED || !op->num_qpts || op->num_qpts == num_qpts, op->ceed, CEED_ERROR_DIMENSION,
+  if (b == CEED_BASIS_COLLOCATED) CeedCall(CeedElemRestrictionGetElementSize(r, &num_qpts));
+  else CeedCall(CeedBasisGetNumQuadraturePoints(b, &num_qpts));
+  CeedCheck(!op->num_qpts || op->num_qpts == num_qpts, op->ceed, CEED_ERROR_DIMENSION,
             "Basis with %" CeedInt_FMT " quadrature points incompatible with prior %" CeedInt_FMT " points", num_qpts, op->num_qpts);
```

Comment:
Not sure about your thoughts/philosophy on error messages, but I think something like:

> "Basis must have the same number of quadrature points ( CeedInt_FMT ) as previously added Bases ( CeedInt_FMT)".

makes more sense to me. I think the term "prior" in particular confused me when I initially saw the error message ("Prior what?").

Also, should we document (if it isn't already) that `CEED_BASIS_COLLOCATED` is treated as if `num_qpts == elem_size` of it's corresponding restriction?